automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java

```diff
@@ -73,13 +74,20 @@ public void sampleOccurred(SampleEvent sampleEvent) {
                                             result.getResponseMessage().replaceAll("\"", "\\\""),
                                             "\", \"Sampler Data\": \"",
                                             samplerData.replaceAll("\"", "\\\""), "\"}");
-            // Persist result to the database
-            testCaseUOW.persistTestCase(result.getSampleLabel(), testScenario, result.isSuccessful(), failureMessage);
+            // Add result to concurrent list
+            TestCase testCase = new TestCase();
```

Comment:
As stated in the above comment, we can set the test cases directly in the test scenario. We might need to use a `CopyOnWriteArrayList` in the test scenario

Reply:
This would be resolved in #474